docs/parser_plugins.md

```diff
@@ -1 +1,176 @@
+In practice, however, parser developers have to manage discrepancies in semantics, shape, data type or units.
+This has lead to five distinct categories of responsibility for the developer to manage:
+
+1. select the relevant files and read them into Python.
+2. target and extract relevant data fields within the source file format on a per-file basis.
+3. match the data of interest with their counterparts in the target/NOMAD schema.
+4. reshape and mangle the data to match the target/NOMAD `Quantity`s' specification. This may include computing derived properties not present in the original source files.
+5. build up a Python `archive` object using the classes provided by the target/NOMAD schema.
+
+In the past, we encountered cases where the delineation between these responsibilities blurred, resulting in widely different parser designs.
+Especially when dealing with the larger, more feature-rich parsers, these designs could become quite complex.
+
+During refactoring, we therefore addressed these responsibilities on a more individual basis.
+This lead to a clear defined _interfaces_ and powerful _tools_ for the developer.
+These are further complimented by _best practices_ and _design protocols_.
+
+More specifically, point 1 (file selection) is now handled by the `MatchingParser` class, from which the parser class inherits. It coordinates with the NOMAD base to scan the uploaded folder tree and select the relevant files.
+
+Point 4 (data mangling) is covered by the NOMAD schema, as you saw in the previous part.
+Not only does this allow for a more consistent handling of the data, it also cleanly partitions the work between parser and schema developers. %% announce Schema Extension?
+In principle, a parser should at most perform trivial manipulations, such as slicing an array or multiplying two extracted values.
+More on the exact "how" later.
+
+Lastly, we will cover tools for point 5 (building the `archive`) and 2 (targeting data fields) in the next subsections.
+Point 5, in particularl, was a developing nuisance, as it required the reconstruction of the already defined target/NOMAD schema.
+That comes across as doing double work: running over the schema once to define it, and a second time to initialize it.
+The main hurdle to automation here was the variable structure of the source data:
+a well-defined hierarchical format does not ensure that individual file structures remains constant.
+Some leaf nodes or branches may simply not be present.
+
+With all of these points covered, developing a parser now comes down to just implementing point 3 (matching source with target).
+This is the stage where data actually becomes semantically enriched and standardized.
+This is a profoundly human task (barring any further leaps in LLM capabilities), where domain expertise is most needed.
+It is thus this scientific curation task that is now most emphasized in parser development, rather than the coding aspects.
```

Comment:
> It is thus this scientific curation task that is now most emphasized in parser development, rather than the coding aspects.

Therefore, this scientific curation task is now most emphasized in parser development.